darts/utils/data/inference_dataset.py

```diff
@@ -168,39 +178,57 @@ def __getitem__(
         ]
 
         # optionally, extract covariates
-        cov_past, cov_future = None, None
+        past_covariate, future_covariate = None, None
         covariate_series = None if self.covariates is None else self.covariates[idx]
         if covariate_series is not None:
             # get start and end indices (integer) of the covariates including historic and future parts
-            cov_start, cov_end = self._covariate_indexer(
-                ts_idx=idx,
+            covariate_start, covariate_end = self._covariate_indexer(
+                target_idx=idx,
                 target_series=target_series,
                 covariate_series=covariate_series,
-                cov_type=self.covariate_type,
+                covariate_type=self.covariate_type,
                 input_chunk_length=self.input_chunk_length,
                 output_chunk_length=self.output_chunk_length,
                 n=self.n,
             )
 
             # extract covariate values and split into a past (historic) and future part
             covariate = covariate_series.random_component_values(copy=False)[
-                cov_start:cov_end
+                covariate_start:covariate_end
             ]
             if self.input_chunk_length != 0:  # regular models
-                cov_past, cov_future = (
+                past_covariate, future_covariate = (
                     covariate[: self.input_chunk_length],
                     covariate[self.input_chunk_length :],
                 )
             else:  # regression ensemble models have a input_chunk_length == 0 part for using predictions as input
-                cov_past, cov_future = covariate, covariate
+                past_covariate, future_covariate = covariate, covariate
 
             # set to None if empty array
-            cov_past = cov_past if cov_past is not None and len(cov_past) > 0 else None
-            cov_future = (
-                cov_future if cov_future is not None and len(cov_future) > 0 else None
+            past_covariate = (
+                past_covariate
+                if past_covariate is not None and len(past_covariate) > 0
+                else None
+            )
+            future_covariate = (
+                future_covariate
+                if future_covariate is not None and len(future_covariate) > 0
+                else None
             )
 
-        return past_target, cov_past, cov_future, target_series
+        # TODO: we need think about the dimensionality of static covariates
+        static_covariate = (
```

Comment:
Sounds good. I checked and `to_numpy(copy=False)` returns a view only when all values/variable have the same dtype in the DataFrame. Otherwise, it returns a copy.